__tests__/app.test.js

```diff
@@ -104,3 +104,69 @@ describe('GET /api', () => {
 
   
   
+describe('POST /api/articles/:article_id/comments', () => {
+    test('should respond with the posted comment object', () => {
+        const comment = {
+            username: 'butter_bridge',
+            body: 'I am commenting for Eric!'
+        }
+
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(comment)
+        .expect(201)
+        .then(({body}) => {
+            const {comment} = body
+                expect(comment).toMatchObject({
+                     author: expect.any(String),
+                     body: expect.any(String),
+                     votes: expect.any(Number),
+                     article_id: 1,
+                     created_at: expect.any(String),
+                     comment_id: expect.any(Number)
+                })
+        })
+    })
+
+    test('POST:400 sends an appropriate status and error message when given a non-existent article_id', () => {
+        const comment = {
+            username: 'butter_bridge',
+            body: 'I am commenting for Eric!'
+        }
+
+        return request(app)
+          .post('/api/articles/nonExistenArticleId/comments')
+          .send(comment)
+          .expect(400)
+          .then(({body}) => {
+            expect(body.msg).toBe('Bad Request');
+          });
+      });
+
+    test('POST:400 sends an appropriate status and error message when given an invalid comment', () => {
+        const comment = {}
+
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(comment)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Bad Request')
+        })
+    });
+
+    test('POST:400 sends an appropriate status and error message when given a valid a non-existent article_id', () => {
+        const comment = {
+            username: 'butter_bridge',
+            body: 'I am commenting for Eric!'
+        }
+
+        return request(app)
+          .post('/api/articles/999/comments')
+          .send(comment)
+          .expect(400)
```

Comment:
This should be a 404 rather than a 400 as it's valid but non-existant

Reply:
alright. thanks katherine

Reply:
done and passing...will go ahead and resolve conflicts and merge now. thanks